docs/_posts/2019-10-21-megabus.md

```diff
@@ -0,0 +1,42 @@
+---
+layout: post
+title: "Megabus API"
+date: 2019-10-21
+excerpt: "EmoDB Megabus"
+tags: [Megabus, Kafka, EmoDB]
+type: [blog]
+---
+
+Megabus
+=======
+
+The Megabus provides a Kafka stream of ordered updates akin to the Databus, but using Apache Kafka topics to deliver updates. The API is Kafka itself. To use the Megabus, you'll discover Kafka, and then use standard Kafka clients/API's to poll and consume records from a topic.
+
+Currently, there is only one "master" Kafka topic for consumption. In the future, we will move to a model where you (1) request your own topic (with a filter for the content you care about), (2) authenticate with Kafka that you do, indeed, have access to consume this topic, and finally (3) consume your topic at the alottted rate.
+
+Discovery
+---------
+
+The Megabus Kafka topic is named based on the environment you choose. The pattern is `<universe>.emodb.<region>.megabus`.
+
+Now that you know the topic you want to poll, you'll need to discover the Kafka brokers in order to connect and begin polling. You can find Kafka via DNS. There are currently 3 Kafka brokers. It's recommended to have at least 2 of them in your `bootstrapServers` Kafka client configuration. The pattern is
+
+    legion-kafka-1.<vpc>.<region>.nexus.bazaarvoice.com:9093,legion-kafka-2.<vpc>.<region>.nexus.bazaarvoice.com:9093,legion-kafka-3.<vpc>.<region>.nexus.bazaarvoice.com:9093
+
+Client
+------
+
+You can use [any Kafka client](https://cwiki.apache.org/confluence/display/KAFKA/Clients), but a Java client is currently recommended due to the compression configuration used, which may not be supported by all clients.
+
+API
+---
+
+Since the API is actually Kafka, you have access to the following Kafka API's:
+
+- [Consumer API](https://kafka.apache.org/documentation/#consumerapi)
+- [Connect API](https://kafka.apache.org/documentation/#connectapi) (recommended for integrating with other systems on raw data)
+- Legion (recommended for all other use-cases, including data augmentation, joins, computations on the stream, and more)
+
+### What about Kafka Streams?
+
+It's our intention that applications that perform computations whose output is destined to go back into Kafka leverage Legion. Basically, if you have any kind of data pipeline involving any sort of computation or  data augmentation in the pipeline, you should be talking to the Legion team. If you want to consume raw EmoDb data in real-time and publish the output to another system entirely, then prefer the Connect API. Although you _could_ use some Kafka Streams API's (any API's that, under the hood, use the Consumer API), we don't recommend it, as you will not be granted permission to write back to Kafka. 
```

Comment:
This should likely be closed source.

Reply:
yeah, as I was writing I realized that probably we have to have some way to make this  internal-only. I guess a Confluence doc would be fine.

Reply:
OK, I put it in confluence here: https://bits.bazaarvoice.com/confluence/pages/viewpage.action?pageId=120339729 and I'm removing this file from the commit.